controllers/database/readiness.go

```diff
@@ -0,0 +1,152 @@
+	case "external":
+		client, err = postgre.GetExternalDatabaseInfo()
+	case "inCluster":
+		//client, err = postgre.GetInClusterRedisInfo()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	defer client.Close(postgre.CXT)
+
+	if err := client.Ping(postgre.CXT); err != nil {
+		postgre.Log.Error(err, "Fail to check Database.", "namespace", postgre.Namespace, "name", postgre.Name)
+		return err
+	}
+	postgre.Log.Info("Database already ready.", "namespace", postgre.Namespace, "name", postgre.Name)
+
+	for _, component := range components {
+		if err := postgre.DeployComponentSecret(component); err != nil {
+			return err
+		}
+	}
+
+	postgre.CRStatus = lcm.New(goharborv1.DatabaseReady).
+		WithStatus(corev1.ConditionTrue).
+		WithReason("database already ready").
+		WithMessage("harbor component database secrets are already create.").
+		WithProperties(*postgre.Properties)
+	return nil
+}
+
+// DeployComponentSecret deploy harbor component database secret
+func (postgre *PostgreSQLReconciler) DeployComponentSecret(component string) error {
+	secret := &corev1.Secret{}
+	secretName := fmt.Sprintf("%s-database", component)
+	propertyName := fmt.Sprintf("%sSecret", component)
+	sc := postgre.generateHarborDatabaseSecret(secretName)
+
+	if err := controllerutil.SetControllerReference(postgre.HarborCluster, sc, postgre.Scheme); err != nil {
+		return err
+	}
+	err := postgre.Client.Get(types.NamespacedName{Name: secretName, Namespace: postgre.Namespace}, secret)
+	if err != nil && kerr.IsNotFound(err) {
+		postgre.Log.Info("Creating Harbor Component Secret",
+			"namespace", postgre.Namespace,
+			"name", secretName,
+			"component", component)
+		err = postgre.Client.Create(sc)
+		if err != nil {
+			return err
+		}
+		postgre.Properties = postgre.Properties.New(propertyName, secretName)
+	} else if err != nil {
+		return err
+	}
+
+	postgre.Properties = postgre.Properties.New(propertyName, secretName)
+
+	return nil
+}
+
+// GetExternalDatabaseInfo returns external database connection client
+func (postgre *PostgreSQLReconciler) GetExternalDatabaseInfo() (*pgx.Conn, error) {
+	var (
+		connect *Connect
+		client  *pgx.Conn
+		err     error
+	)
+	spec := postgre.HarborCluster.Spec.Database.Spec
+	if spec.SecretName == "" {
+		return nil, errors.New(".database.spec.secretName is invalid")
+	}
+
+	if spec.SecretName != "" {
+		connect, err = GetExternalDatabaseConn(spec, postgre.Namespace, postgre.Client)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	postgre.Connect = connect
+
+	url := connect.GenDatabaseUrl()
+
+	client, err = pgx.Connect(postgre.CXT, url)
+	if err != nil {
+		postgre.Log.Error(err, "Unable to connect to database")
+		return nil, err
+	}
+
+	return client, nil
+}
+
+// GetExternalDatabaseConn returns external database connection info
+func GetExternalDatabaseConn(spec *goharborv1.PostgresSQL, namespace string, client k8s.Client) (*Connect, error) {
+	external := &PostgreSQLReconciler{
```

Comment:
Why a PostgreSQLReconciler need to be created? 

Just for calling `GetDatabaseConn`？

Reply:
I need both inCluster and External to call the `GetDatabaseConn` function, so here I need to create a PostgreSQLReconciler.